k2/csrc/fsa.h

```diff
@@ -148,15 +144,43 @@ struct DenseFsa {
       CAUTION: we may later enforce that stride == num_symbols, in order to
       be able to know the layout of a phantom matrix of arcs.  (?)
    */
-  DenseFsa(Weight *data, int32_t T, int32_t num_symbols, int32_t stride);
+  DenseFsa(float *data, int32_t T, int32_t num_symbols, int32_t stride);
 };
 
 struct Fst {
   Fsa core;
   std::vector<int32_t> aux_label;
 };
 
-using StatePair = std::pair<StateId, StateId>;
+/*
+  This demonstrates an interface for a deterministic FSA or FST; it's similar
+  to Kaldi's DeterministicOnDemandFst class.  It can be used for things like
+  language models.  Actually we'll template on types like this.  There is no
+  need to actually inherit from this class.  */
+class DeterministicGenericFsa {
+
+  int32_t Start();
+
+
+  bool LookupArc(int32_t cur_state,
```

Comment:
Why are there no `const` modifiers for `Get` ?

Reply:
Because this is an interface for a possibly dynamic object.